examples/scripts/deconv_modl_train_foam.py

```diff
@@ -0,0 +1,348 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This file is part of the SCICO package. Details of the copyright
+# and user license can be found in the 'LICENSE.txt' file distributed
+# with the package.
+
+r"""
+Training of MoDL for Deconvolution
+==================================
+
+This example demonstrates the training and application of a model-based deep learning
+(MoDL) architecture described in :cite:`aggarwal-2019-modl`
+ for a deconvolution (deblurring) problem.
+
+The source images are foam phantoms generated with xdesign.
+
+A class [flax.MoDLNet](../_autosummary/scico.learning.rst#scico.learning.MoDL)
+ implements the MoDL architecture, which solves the optimization problem
+
+  $$\mathrm{argmin}_{\mathbf{x}} \; \| A \mathbf{x} - \mathbf{y} \|_2^2 + \lambda \, \| \mathbf{x} - \mathrm{D}_w(\mathbf{x})\|_2^2 \;,$$
+
+where $A$ is a circular convolution, $\mathbf{y}$ is a set of blurred images, $\mathrm{D}_w$ is the
+ regularization (a denoiser), and $\mathbf{x}$ is the set of deblurred images.
+  The MoDL abstracts the iterative solution by an unrolled network where each iteration corresponds
+  to a different stage in the MoDL network and updates the prediction by solving
+
+  $$\mathbf{x}^{k+1} = (A^T A + \lambda \, \mathbf{I})^{-1} (A^T \mathbf{y} + \lambda \, \mathbf{z}^k) \;,$$
```

Comment:
Consistency: I don't think we use mathbf for an operator anywhere else.

Reply:
Removed \mathbf{I}